File: RockitCompiler/benchmarks/bench_parse_int.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	seed := int64(42)
	sum := int64(0)
	valid := 0
	invalid := 0
	for i := 0; i < 1000000; i++ {
		seed = (seed * 1103515245 + 12345) % 2147483648
		var s string
		if seed%5 == 0 {
			s = "abc"
		} else {
			s = strconv.FormatInt(seed%1000000, 10)
		}
		v, err := strconv.Atoi(s)
		if err != nil {
			invalid++
		} else {
			sum += int64(v)
			valid++
		}
	}
	fmt.Printf("%d %d %d\n", sum, valid, invalid)
}

File: RockitCompiler/benchmarks/bench_parse_int.cpp
#include <cstdio>
#include <cstdint>
#include <cstdlib>
#include <string>

int main() {
    int64_t seed = 42;
    int64_t sum = 0;
    int64_t valid = 0;
    int64_t invalid = 0;
    for (int i = 0; i < 1000000; i++) {
        seed = (seed * 1103515245 + 12345) % 2147483648LL;
        std::string s;
        if (seed % 5 == 0) {
            s = "abc";
        } else {
            s = std::to_string(seed % 1000000);
        }
        char* end;
        int64_t v = strtoll(s.c_str(), &end, 10);
        if (end != s.c_str() && *end == '\0') {
            sum += v;
            valid++;
        } else {
            invalid++;
        }
    }
    printf("%lld %lld %lld\n", sum, valid, invalid);
    return 0;
}
